### PR TITLE
feat(celery): Emit a counter when a pool process starts

### PR DESCRIPTION
**Pull request by untitaker**
https://docs.celeryq.dev/en/stable/userguide/signals.html#worker-process-init

ref: INC-603

### CODE REVIEW

**Review by mitsuhiko — APPROVED** [review]

File: src/sentry/celery.py [hunk 1]
```diff
@@ -101,6 +101,11 @@ class SentryCelery(Celery):
     task_cls = SentryTask
 
 
+@signals.worker_process_init.connect
+def record_worker_init(*args, **kwargs):
+    metrics.incr("sentry.jobs.process.start")
```
Comment (fpacifici) [inline src/sentry/celery.py:106]: You may want to add a tag to identify the worker

Comment (untitaker) [reply src/sentry/celery.py:106]: replied in slack, we should ahve `kube_deployment` and whatever global app tags are registered

**Comment by mwarkentin** [comment]
We've been seeing some [WorkerLostError](https://sentry.sentry.io/issues/4838063708/?referrer=slack) that seem to have started when this change went out. The number of errors escalated significantly today (enough to trigger CrashLoop pages for on-call).

image

**Comment by mwarkentin** [comment]
Reverted in https://github.com/getsentry/sentry/pull/64102

**Pull request merged**